docs/user-guide/continous-development.md

```diff
@@ -0,0 +1,103 @@
+---
+description: How to use Continuous Development in Elastisys Compliant Kubernetes, the security-focused Kubernetes distribution.
+---
+
+# Continuous Development
+
+When developing on Kubernetes, it can be time-consuming to manually run commands to build new images,
+push images to a container registry, update Kubernetes manifests, and deploy new manifests to the cluster with
+each source-code change.
+
+[Skaffold](https://github.com/GoogleContainerTools/skaffold) is a tool that can be used to ease this process.
+Skaffold will automate the process of building, pushing and deploying new images based on changes to the source-code.
+
+Skaffold requires minimal setup as the tool has no cluster-side components, and will automatically detect the configuration to use.
+
+## Installing Skaffold
+
+The Skaffold CLI can be installed by downloading and installing the latest release. Instructions can be found in the [Skaffold
+documentation](https://skaffold.dev/docs/install/):
+
+```sh
+curl -Lo skaffold https://storage.googleapis.com/skaffold/releases/v2.0.1/skaffold-linux-amd64 && \
+sudo install skaffold /usr/local/bin/
+```
+
+## Getting started with Skaffold
+
+!!!note
+    Skaffold will use the active `KUBECONFIG` to authenticate to the Kubernetes cluster.
+
+If you havent't done so already, clone the user demo:
+
+```bash
+git clone https://github.com/elastisys/compliantkubernetes/
+cd compliantkubernetes/user-demo
+```
+
+<!-- TODO: I didn't read the docs thingie? -->
+
+### Initialize Skaffold
+
+```bash
+skaffold init
+```
+
+This command will scan the current project for images to build and Kubernetes manifests to deploy. For each image that
+Skaffold finds you will be prompted to specify how to build them, or if they are not built by the current project.
+
+The first image skaffold finds is busybox, however this image is not built from this project, so choose
+`None (image not built from these sources)`
+
+![Skaffold init Output1](/compliantkubernetes/user-guide/img/skaffold-busybox.png)
+
+The second image Skaffold finds is the user-demo image and this image is built from the Dockerfile.
+
+![Skaffold init Output2](/compliantkubernetes/user-guide/img/skaffold-user-demo.png)
+
+Skaffold then asks for which resources we want to create Kubernetes resources, but as the image already has a helm-chart
+this can be skipped by pressing enter.
+
+Skaffold will then create the `skaffold.yaml` file containing our configuration. Skaffold will also automatically detect the
+helm-chart that deploys the `user-demo` image.
+
+### Developing
+
+To start developing using Skaffold run:
+
+```bash
+skaffold dev
```

Comment:
I got the following error when trying to start developing with Skaffold: `Build Failed. No push access to specified image repository. Try running with --default-repo flag.`

Full output:
```skaffold dev
Listing files to watch...
 - i-didnt-read-the-docs/ck8s-user-demo
Generating tags...
 - i-didnt-read-the-docs/ck8s-user-demo -> i-didnt-read-the-docs/ck8s-user-demo:b8ff4f2-dirty
Checking cache...
 - i-didnt-read-the-docs/ck8s-user-demo: Not found. Building
Starting build...
Building [i-didnt-read-the-docs/ck8s-user-demo]...
Target platforms: [linux/amd64]
#1 [internal] load build definition from Dockerfile
#1 sha256:eaf635020728c1bb3ce56721f9c6d141ab818154627ed8b1cf6f0c34a26b5392
#1 transferring dockerfile: 344B 0.0s done
#1 DONE 0.1s

#2 [internal] load .dockerignore
#2 sha256:fb25e1b0cb46a747ce48f77ce0b3caf27d9e453bc7cebc9a5da5e9f72788667d
#2 transferring context: 73B done
#2 DONE 0.1s

#3 [internal] load metadata for docker.io/library/node:14-alpine
#3 sha256:a4d7d3caeb58aee7272d38e41f54c299e593ee826f0d11b0b153dd03a81fefe1
#3 ...

#4 [auth] library/node:pull token for registry-1.docker.io
#4 sha256:4a8f61feea5c57684b223c0e966f2205d7082f90df0e0e98303d8ab4767ffee2
#4 DONE 0.0s

#3 [internal] load metadata for docker.io/library/node:14-alpine
#3 sha256:a4d7d3caeb58aee7272d38e41f54c299e593ee826f0d11b0b153dd03a81fefe1
#3 DONE 1.5s

#5 [1/5] FROM docker.io/library/node:14-alpine@sha256:12b14bdfa8c89a1a060c53b5714157085700660b12ab7c50a907a4e19d95b6bf
#5 sha256:739b88491321193e71cf05769c309efadb0c00fefae572cb60184f560c0f3020
#5 resolve docker.io/library/node:14-alpine@sha256:12b14bdfa8c89a1a060c53b5714157085700660b12ab7c50a907a4e19d95b6bf 0.0s done
#5 ...

#7 [internal] load build context
#7 sha256:7c44eafc66e8535fac9e561def759607215e1d07e9c64a9b22b333d5d90afbb9
#7 transferring context: 44.28kB done
#7 DONE 0.1s

#5 [1/5] FROM docker.io/library/node:14-alpine@sha256:12b14bdfa8c89a1a060c53b5714157085700660b12ab7c50a907a4e19d95b6bf
#5 sha256:739b88491321193e71cf05769c309efadb0c00fefae572cb60184f560c0f3020
#5 sha256:eaed9ed12bae01d0f5473c465f1b50b2ffe2c838c1482d82e4771e54bb62f214 1.16kB / 1.16kB done
#5 sha256:87112681acadc5285d472f954d23560bee9d105e156ca59e48f41bf4bf16881f 6.44kB / 6.44kB done
#5 sha256:ca7dd9ec2225f2385955c43b2379305acd51543c28cf1d4e94522b3d94cce3ce 0B / 2.81MB 0.1s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 0B / 37.72MB 0.1s
#5 sha256:c8d9211bf297cc0df2b07d2ad8ee5a83f5f479165228a1a513004dba925cfc20 0B / 2.37MB 0.1s
#5 sha256:12b14bdfa8c89a1a060c53b5714157085700660b12ab7c50a907a4e19d95b6bf 1.43kB / 1.43kB done
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 2.10MB / 37.72MB 0.4s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 4.19MB / 37.72MB 0.7s
#5 sha256:ca7dd9ec2225f2385955c43b2379305acd51543c28cf1d4e94522b3d94cce3ce 1.05MB / 2.81MB 0.9s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 7.34MB / 37.72MB 1.1s
#5 sha256:c8d9211bf297cc0df2b07d2ad8ee5a83f5f479165228a1a513004dba925cfc20 1.05MB / 2.37MB 1.1s
#5 sha256:ca7dd9ec2225f2385955c43b2379305acd51543c28cf1d4e94522b3d94cce3ce 2.10MB / 2.81MB 1.4s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 9.44MB / 37.72MB 1.4s
#5 sha256:c8d9211bf297cc0df2b07d2ad8ee5a83f5f479165228a1a513004dba925cfc20 2.10MB / 2.37MB 1.4s
#5 sha256:ca7dd9ec2225f2385955c43b2379305acd51543c28cf1d4e94522b3d94cce3ce 2.81MB / 2.81MB 1.5s done
#5 sha256:c8d9211bf297cc0df2b07d2ad8ee5a83f5f479165228a1a513004dba925cfc20 2.37MB / 2.37MB 1.5s done
#5 extracting sha256:ca7dd9ec2225f2385955c43b2379305acd51543c28cf1d4e94522b3d94cce3ce 0.1s
#5 sha256:14523e6c1be223f204f3266fd2d66cc5d9caf28f41364bfa565923c09ed3b4db 0B / 449B 1.6s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 11.53MB / 37.72MB 1.7s
#5 extracting sha256:ca7dd9ec2225f2385955c43b2379305acd51543c28cf1d4e94522b3d94cce3ce 0.1s done
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 14.68MB / 37.72MB 1.9s
#5 sha256:14523e6c1be223f204f3266fd2d66cc5d9caf28f41364bfa565923c09ed3b4db 449B / 449B 1.7s done
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 16.78MB / 37.72MB 2.1s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 18.87MB / 37.72MB 2.3s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 20.97MB / 37.72MB 2.5s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 24.12MB / 37.72MB 2.7s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 27.26MB / 37.72MB 3.0s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 29.36MB / 37.72MB 3.2s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 32.51MB / 37.72MB 3.5s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 34.60MB / 37.72MB 3.7s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 36.70MB / 37.72MB 3.8s
#5 sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 37.72MB / 37.72MB 3.9s done
#5 extracting sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 0.1s
#5 extracting sha256:f6c774934c91f00c542a1fb93212d42cf819d47de47169f00eba8db7e7026f51 0.8s done
#5 extracting sha256:c8d9211bf297cc0df2b07d2ad8ee5a83f5f479165228a1a513004dba925cfc20 0.1s
#5 extracting sha256:c8d9211bf297cc0df2b07d2ad8ee5a83f5f479165228a1a513004dba925cfc20 0.2s done
#5 extracting sha256:14523e6c1be223f204f3266fd2d66cc5d9caf28f41364bfa565923c09ed3b4db done
#5 DONE 5.3s

#6 [2/5] WORKDIR /home/node/app
#6 sha256:03d6c5ea82fffb75bca94f124106f41c837bb71b9f5bb6f4ff7402ef2d2372c7
#6 DONE 0.2s

#8 [3/5] COPY package*.json ./
#8 sha256:36353eb8586c10098b157eaa763431762dda387b6c4ad8a0df91f7b07de73eba
#8 DONE 0.1s

#9 [4/5] RUN npm install
#9 sha256:94a71e70e6a63b5f55d05869f1867c4a39f4841c2ab73a0d19cc09230ca8d54a
#9 3.121 added 112 packages from 141 contributors and audited 116 packages in 2.417s
#9 3.170 
#9 3.170 3 packages are looking for funding
#9 3.170   run `npm fund` for details
#9 3.170 
#9 3.171 found 0 vulnerabilities
#9 3.171 
#9 DONE 3.3s

#10 [5/5] COPY . ./
#10 sha256:235ab4bc8e9829747fb644e64246c4ff920b8af39586bad0f865f62dc94386c5
#10 DONE 0.1s

#11 exporting to image
#11 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
#11 exporting layers
#11 exporting layers 0.4s done
#11 writing image sha256:d580d9c623d1cc7ed8c173dfb94068a9158d2afd14fe804c98ff5aa75da63d68 done
#11 naming to docker.io/i-didnt-read-the-docs/ck8s-user-demo:b8ff4f2-dirty done
#11 DONE 0.5s
The push refers to repository [docker.io/i-didnt-read-the-docs/ck8s-user-demo]
03ae1671dd46: Preparing
1bb5b46a233a: Preparing
58cecff8121c: Preparing
305b85ac6849: Preparing
3af570d59c20: Preparing
58db50de30fc: Preparing
d28200441f34: Preparing
e5e13b0c77cb: Preparing
58db50de30fc: Waiting
e5e13b0c77cb: Waiting
d28200441f34: Waiting
Build Failed. No push access to specified image repository. Try running with `--default-repo` flag.```

Reply:
Yes, sorry, forgot about this, it is because the demo-project has specified the image 'i-didnt-read-the-docs'

I added some changes to highlight the changes needed in the values.yaml file of the helm chart